sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,463 @@
++Callable+ `f` and by a list of properties `propertiesT`.
+The +Callable+ `f` defines the asynchronous task to be
+associated with the `task_sequence`, and requiring an auto reference ensures
+that each `f` be statically resolvable at compile time.  Static resolvability by
+the compiler is desirable when compiling for spatial architectures as it can
+enable the generation of more efficient hardware. Furthermore, the partial
+specialization `typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...)`
+helps make it easier to write the class member functions (eg: `get`).
+
+The `task_sequence` interface consists of two member functions, `async` and
+`get`. The `async` function asynchronously invokes `f`. By calling `async` on a
+`task_sequence` more than once, the user implies that the invocations of `f` can
+be run in parallel. The implementation is, however, not obligated to run these
+invocations in parallel except in so far as is necessary to meet the forward
+progress guarantees outlined in the section on Progress Guarantees.
+
+The `async` function call is non-blocking in that it may return before the
+asynchronous `f` invocation completes executing, and potentially before `f` even
+begins executing (return from the `async` provides no implicit information on
+the execution status of `f`).
+
+The `get` member function retrieves results in the same order in which
+`async` was invoked. The call blocks if there is no result to retrieve. Both
+functions may only be invoked on the device on which a `task_sequence` object
+has been instantiated. Calling `async` or `get` on a different device results in
+undefined behavior.
+
+[frame="topbot",options="header"]
+|===
+|Functions |Description
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+task_sequence(const propertyListT &);
+
+task_sequence();
+----
+|
+Constructor for `task_sequence`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+void async(ArgsT... args);
+----
+|
+Asynchronously calls `f` with `args`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+ReturnT get();
+----
+|
+Synchronously retrieves the result of an `async` call.
+
+Results are retrieved in the order in which the `async` calls were made.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+~task_sequence();
+----
+|
+Destructor for `task_sequence`.
+
+Implicitly invokes `get` on all outstanding invocations launched through `async
+unless this `task_sequence` object was instantiated with the `balanced` property
+defined below.
+|===
+
+=== `task_sequence` Scoping
+
+`task_sequence` objects should retire all outstanding `async` invocations before
+exiting scope - this is performed by the `task_sequence` destructor unless the
+`balanced` property was specified in which case it is the programmer's
+responsibility to ensure that `get` has been called for all invocations launched
+using `async`, before the lifetime of the `task_sequence` object ends.
+
+A count of outstanding `async` invocations could be maintained by incrementing a
```

Comment:
I think this paragraph is unnecessary.  It's purely an implementation detail and irrelevant to the user.

Reply:
Removed.